lib/elixir_analyzer/exercise_test/common_checks/compiler_warnings.ex

```diff
@@ -12,8 +12,10 @@ defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.CompilerWarnings do
     warnings =
       capture_io(:stderr, fn ->
         try do
-          Code.compile_quoted(code_ast, Path.basename(code_path))
-          |> Enum.each(fn {module, _binary} ->
+          :elixir_compiler.quoted(code_ast, code_path, fn _, _ -> :ok end)
+          |> Enum.each(fn {module, map, binary} ->
+            Module.ParallelChecker.verify([{map, binary}], [])
+
```

Comment:
With the hints from https://elixirforum.com/t/why-does-code-compile-string-report-deprecation-warnings-only-sometimes/44012/2 I came up with this solution. 

I took the compilations step from `compile_quoted`:
```
  def compile_quoted(quoted, file \\ "nofile") when is_binary(file) do
    loaded = :elixir_compiler.quoted(quoted, file, fn _, _ -> :ok end)
    Enum.map(loaded, fn {module, _map, binary} -> {module, binary} end)
  end
```

and then the verification step from `compile_file`
```elixir
  def compile_file(file, relative_to \\ nil) when is_binary(file) do
    loaded = :elixir_compiler.file(find_file(file, relative_to), fn _, _ -> :ok end)
    verify_loaded(loaded)
  end
```
```elixir
  defp verify_loaded(loaded) do
    maps_binaries = Enum.map(loaded, fn {_module, map, binary} -> {map, binary} end)
    Module.ParallelChecker.verify(maps_binaries, [])
    Enum.map(loaded, fn {module, _map, binary} -> {module, binary} end)
  end
```

I'm not sure if it's a great idea 🤔 the whole module `Module.ParallelChecker` is supposed to be private...

Reply:
I don’t think it’s a good idea to rely on private modules… I found this thread the other day where José Valim was urging users not to do that, as it makes changing internal logic a lot more complex if people rely on it (I’m on my phone right now, I can find the link later)

instead, why don’t we compile the file directly? The part should be in the source. 
For testing, we could simply write a temp file at the same time as the source struct is being built.

Reply:
Mmh maybe for testing it’s not that simple actually… maybe we can do that particular test from files only then?

Reply:
Ah, it looks like I don't need to find the link I was talking about since José answered the same thing to your thread :)

Reply:
Well, to me "private" code is more like social contract, and my interpretation of that contract is that I shouldn't use them, but if I really really have to, I have no right to complain when something unexpected happens 🙂. I opened a bug report for Elixir about this, but we don't need to wait for a fix because using `Code.compile_file` was trivial.